src/plugins/faultManagement/FaultManagementPlugin.js

```diff
@@ -0,0 +1,33 @@
+import FaultManagementViewProvider from './FaultManagementViewProvider';
+import FaultManagementObjectProvider from './FaultManagementObjectProvider';
+import FaultManagementInspectorViewProvider from './FaultManagementInspectorViewProvider';
+
+import { FAULT_MANAGEMENT_TYPE, FAULT_MANAGEMENT_NAMESPACE } from './constants';
+
+export default function FaultManagementPlugin(config = {}) {
+    return function (openmct) {
+        openmct.types.addType(FAULT_MANAGEMENT_TYPE, {
+            name: 'Fault Management',
+            creatable: false,
+            description: 'Fault Management View',
+            cssClass: 'icon-telemetry'
+        });
+
+        openmct.objectViews.addProvider(new FaultManagementViewProvider(openmct));
+        openmct.inspectorViews.addProvider(new FaultManagementInspectorViewProvider(openmct));
+        openmct.objects.addProvider(FAULT_MANAGEMENT_NAMESPACE, new FaultManagementObjectProvider(openmct));
+
+        openmct.faults.setConfig(config);
+        if (config.historicaFaultProvider) {
```

Comment:
See previous comments, for consistency let's just make this a single fault provider. Although as I say, this approach is interesting and we should consider it for the future.